File: backend/app/schemas/document.py
"""Pydantic schemas for Document Drafting and Editing."""

from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field


class DocumentSection(BaseModel):
    id: str
    title: str
    content: str


class DocumentResponseData(BaseModel):
    document_id: str
    case_id: str
    type: str
    title: str
    sections: List[DocumentSection]
    quality_score: float = 8.0
    quality_warnings: List[str] = Field(default_factory=list)
    disclaimer: str
    created_at: str


class UpdateDocumentRequest(BaseModel):
    title: Optional[str] = None
    sections: List[DocumentSection]

File: frontend/src/types/document.ts
export type DocumentType =
  | "complaint"
  | "consumer_complaint"
  | "labor_complaint"
  | "tenant_notice"
  | "legal_notice"
  | (string & {});

export interface DocumentSection {
  id: string;
  title: string;
  content: string;
}

/** POST /api/cases/{id}/document · GET/PUT /api/documents/{id} */
export interface DocumentData {
  document_id: string;
  case_id: string;
  type: DocumentType;
  title: string;
  sections: DocumentSection[];
  quality_score: number;
  quality_warnings: string[];
  disclaimer: string;
  created_at: string;
}

export interface UpdateDocumentRequest {
  title?: string;
  sections: DocumentSection[];
}
